tests/trainer/test_optimizers.py

```diff
@@ -309,12 +347,105 @@ def test_configure_optimizers_with_frequency(tmpdir):
     model = EvalModelTemplate()
     model.configure_optimizers = model.configure_optimizers__multiple_optimizers_frequency
 
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        max_epochs=1
-    )
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
+    result = trainer.fit(model)
+    assert result
+
+
+def test_configure_optimizers_with_frequency_and_lr_schedulers(tmpdir):
+    """
+    Test that opt_idx is set for LR schedulers when corresponding frequency is set for multiple optimizers.
+    """
+    model = BoringModel()
+    model.configure_optimizers = lambda: [
+        {'optimizer': optim.Adam(model.parameters(), lr=0.01), 'frequency': 5},
+        {
+            'optimizer': optim.Adam(model.parameters(), lr=0.01),
+            'frequency': 1,
+            'lr_scheduler': {
+                'scheduler': optim.lr_scheduler.OneCycleLR(
+                    optim.Adam(model.parameters(), lr=0.01), max_lr=0.01, total_steps=1
+                ),
+                'interval': 'step',
+            },
+        },
+    ]
+
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
+    _, lr_schedulers, _ = trainer.init_optimizers(model)
+    assert lr_schedulers[0]['opt_idx'] == 1
+
+
+def test_step_scheduling_for_multiple_optimizers_with_frequency(tmpdir):
+    """
+    Test that step LR schedulers for multiple optimizers follow
+    the optimizer frequencies when corresponding frequency is set.
+    """
+
+    class DummyStepModel(BoringModel):
+        def training_step(self, batch, batch_idx, optimizer_idx):
+            output = self.layer(batch)
+            loss = self.loss(batch, output)
+            return {"loss": loss}
+
+        def configure_optimizers(self):
+            optimizer1 = optim.Adam(self.parameters(), lr=0.01)
+            optimizer2 = optim.Adam(self.parameters(), lr=0.01)
+
+            return [
+                {'optimizer': optimizer1, 'frequency': 5},
+                {
+                    'optimizer': optimizer2,
+                    'frequency': 1,
+                    'lr_scheduler': {
+                        'scheduler': optim.lr_scheduler.OneCycleLR(optimizer2, max_lr=0.01, total_steps=1),
+                        'interval': 'step',
+                    },
+                },
+            ]
```

Comment:
```suggestion
            lr_dict = {
                'scheduler': optim.lr_scheduler.OneCycleLR(optimizer2, max_lr=0.01, total_steps=1),
                'interval': 'step',
            }
            return [
                {'optimizer': optimizer1, 'frequency': 5},
                {'optimizer': optimizer2, 'frequency': 1, 'lr_scheduler': lr_dict},
            ]
```
a bit cleaner. mind update it like this everywhere.